### PR TITLE
feat(fa): Allowing admins to create and update api keys for api 

### PR DESCRIPTION
**Pull request by MargretFinnboga**
#Allowing admins to create and update keys

[Viðmót í Veitu fyrir aðgangsupplýsingar](https://app.asana.com/0/1205807691300420/1206261438834065)

## What

- Created endpoint to create, update and fetch keys(keys are encrypted to the DB)
- Added two modal update and create key (and made api settings container for them)
- Added also component to make the content about the api key appear or not appear 
- moved functionality to hooks for refactoring 

## Why

- b

## Screenshots / Gifs
í stillingum 
Screenshot 2024-02-05 at 15 08 44
Screenshot 2024-02-05 at 15 08 49
Screenshot 2024-02-05 at 15 08 56

Modalarnir 
Screenshot 2024-02-05 at 15 09 30
Screenshot 2024-02-05 at 15 09 38

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: apps/financial-aid/web-veita/src/components/MunicipalityAdminSettings/ApiKeyModal/UpdateApiKeyModal.tsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+  useEffect(() => {
+    setApiKeyState({ ...apiKeyState, apiKey: apiKey, name: name })
+  }, [name, apiKey])
+
+  return (
+    <ActionModal
+      isVisible={isVisible}
+      setIsVisible={setIsVisible}
+      header={'Uppfæra lykill'}
```
Comment (raldur) [inline apps/financial-aid/web-veita/src/components/MunicipalityAdminSettings/ApiKeyModal/UpdateApiKeyModal.tsx:40]:
```suggestion
      header={'Uppfæra lykil'}
```

File: apps/financial-aid/web-veita/src/components/MunicipalityAdminSettings/ApiKeyModal/UpdateApiKeyModal.tsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        autoComplete="off"
+        hasError={apiKeyState.hasError && !apiKeyState.name}
+        onChange={(event) => {
+          setApiKeyState({
+            ...apiKeyState,
+            name: event.target.value,
+            hasError: false,
+          })
+        }}
+        errorMessage="Til að uppfæra til lykill þarf nafn að vera til staðar"
```
Comment (raldur) [inline apps/financial-aid/web-veita/src/components/MunicipalityAdminSettings/ApiKeyModal/UpdateApiKeyModal.tsx:69]: > Til að uppfæra til lykill þarf nafn að vera til staðar

image

File: apps/financial-aid/web-veita/src/components/MunicipalityAdminSettings/ApiKeyModal/CreateApiKeyModal.tsx [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+      <Input
+        label="Lykill"
+        name="apiKey"
+        value={apiKeyState?.apiKey}
+        backgroundColor="blue"
+        autoComplete="off"
+        readOnly
+      />
+      <Text marginTop={1} marginBottom={3} variant="small">
+        Lykill hefur verið búin til
```
Comment (raldur) [inline apps/financial-aid/web-veita/src/components/MunicipalityAdminSettings/ApiKeyModal/CreateApiKeyModal.tsx:93]:
```suggestion
        Lykill hefur verið búinn til
```

File: apps/financial-aid/web-veita/src/components/MunicipalityAdminSettings/ApiKeyModal/CreateApiKeyModal.tsx [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+        autoComplete="off"
+        hasError={apiKeyState.hasError && !apiKeyState.name}
+        onChange={(event) => {
+          setApiKeyState({
+            ...apiKeyState,
+            name: event.target.value,
+            hasError: false,
+          })
+        }}
+        errorMessage="Til að búa til lykill þarf nafn að vera til staðar"
```
Comment (raldur) [inline apps/financial-aid/web-veita/src/components/MunicipalityAdminSettings/ApiKeyModal/CreateApiKeyModal.tsx:78]:
```suggestion
        errorMessage="Til að búa til lykil þarf nafn að vera til staðar"
```

**Review by raldur — CHANGES_REQUESTED** [review]
Bara nokkrar innsláttar- eða stafsetningavillur 👨‍🏫

**Review by raldur — APPROVED** [review]
🤌

**Review by Rikkiabb — APPROVED** [review]
Looks good, I am worried though about the design of the keys. That is expirary of them and I would also like it much more if the client wouldn't have to be able to show them (they would only be visibile on creation). But those worries are not about this code but the general design of the system.

**Pull request merged**